codegen/smithy-aws-go-codegen/src/main/java/software/amazon/smithy/aws/go/codegen/AddAwsConfigFields.java

```diff
@@ -0,0 +1,148 @@
+import software.amazon.smithy.go.codegen.GoSettings;
+import software.amazon.smithy.go.codegen.GoWriter;
+import software.amazon.smithy.go.codegen.SymbolUtils;
+import software.amazon.smithy.go.codegen.TriConsumer;
+import software.amazon.smithy.go.codegen.integration.ConfigField;
+import software.amazon.smithy.go.codegen.integration.GoIntegration;
+import software.amazon.smithy.go.codegen.integration.RuntimeClientPlugin;
+import software.amazon.smithy.model.Model;
+import software.amazon.smithy.model.knowledge.ServiceIndex;
+import software.amazon.smithy.model.shapes.ServiceShape;
+import software.amazon.smithy.utils.ListUtils;
+import software.amazon.smithy.utils.SetUtils;
+
+/**
+ * Registers additional AWS specific client configuration fields
+ */
+public class AddAwsConfigFields implements GoIntegration {
+    private static final Logger LOGGER = Logger.getLogger(AddAwsConfigFields.class.getName());
+
+    private static final String REGION_CONFIG_NAME = "Region";
+    private static final String CREDENTIALS_CONFIG_NAME = "Credentials";
+    private static final String ENDPOINT_RESOLVER_CONFIG_NAME = "EndpointResolver";
+    private static final String HTTP_CLIENT_CONFIG_NAME = "HTTPClient";
+    private static final String LOGGER_CONFIG_NAME = "Logger";
+    private static final String LOG_LEVEL_CONFIG_NAME = "LogLevel";
+    private static final String RETRYER_CONFIG_NAME = "Retryer";
+    private static final List<ConfigField> UNIVERSAL_FIELDS = new ArrayList<>(SetUtils.of(
+            ConfigField.builder()
+                    .name(REGION_CONFIG_NAME)
+                    .type(getUniversalSymbol("string"))
+                    .documentation("The region to send requests to. (Required)")
+                    .build(),
+            ConfigField.builder()
+                    .name(ENDPOINT_RESOLVER_CONFIG_NAME)
+                    .type(getAwsCoreSymbol("EndpointResolver"))
+                    .documentation("The resolver to use for looking up endpoints for the service.")
+                    .build(),
+            ConfigField.builder()
+                    .name(RETRYER_CONFIG_NAME)
+                    .type(getAwsCoreSymbol("Retryer"))
+                    .documentation("Retryer guides how HTTP requests should be retried in case of\n"
+                            + "recoverable failures. When nil the API client will use a default\n"
+                            + "retryer.")
+                    .build(),
+            ConfigField.builder()
+                    .name(LOG_LEVEL_CONFIG_NAME)
+                    .type(getAwsCoreSymbol("LogLevel"))
+                    .documentation("An integer value representing the logging level.")
+                    .build(),
+            ConfigField.builder()
+                    .name(LOGGER_CONFIG_NAME)
+                    .type(getAwsCoreSymbol("Logger"))
+                    .documentation("The logger writer interface to write logging messages to.")
+                    .build()
```

Comment:
We should create github issues to track the design ideas we discussed for Logger improvements with the SDK and manager.

Reply:
I've created https://github.com/aws/aws-sdk-go-v2/issues/594 to track this design requirement.